plugins/com.google.cloud.tools.eclipse.appengine.facets.test/src/com/google/cloud/tools/eclipse/appengine/facets/ui/navigator/ConfigurationFileUtils.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.eclipse.appengine.facets.ui.navigator;
+
+import static org.junit.Assert.fail;
+
+import com.google.cloud.tools.eclipse.appengine.facets.WebProjectUtil;
+import com.google.common.base.Strings;
+import java.io.ByteArrayInputStream;
+import java.nio.charset.StandardCharsets;
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.Path;
+
+/** Utility methods for creating App Engine configuration files for testing purposes. */
+public class ConfigurationFileUtils {
+
+  /** WebProjectUtil#createFileInWebInf() does not overwrite files. */
+  public static IFile createInWebInf(IProject project, IPath path, String contents) {
```

Comment:
It would be nice if the method names here and in `WebProjectUtil#createFileInWebInf()` were either consistent or more clearly delineated.  E.g. this one could also be createFileInWebInf or createOrReplaceFileInWebInf

Reply:
Considering that this is only for testing, I've actually thought about making this dead simple, e.g., to just place the file in a specified location which will work whether the project has the WTP facet or not. [`CloudSdkStagingHelperTest`](https://github.com/GoogleCloudPlatform/google-cloud-eclipse/blob/v1.7.0D/plugins/com.google.cloud.tools.eclipse.appengine.deploy.test/src/com/google/cloud/tools/eclipse/appengine/deploy/CloudSdkStagingHelperTest.java#L60) does similar things for testing, and maybe we could reuse code for both (later).

Reply:
They're used for the _Deployment Assembly_ case too.